test/dates/ranges.jl

```diff
@@ -482,10 +482,10 @@ function testmonthranges2(n)
 end
 testmonthranges2(100000)
 
-@test length(Dates.Year(1):Dates.Year(10)) == 10
+@test length(Dates.Year(1):Dates.Year(1):Dates.Year(10)) == 10
```

Comment:
I can see the sense in supporting it when you're typing the endpoints explicitly: I think it's pretty obvious what the user wants with `Minute(1):Minute(60)`. But I worry about code like this:

```julia
function foo(a::Period, b::Period)
    ap, bp = promote(a, b)
    ap:bp
end
```
where now knowing what you'll get relies on a fair amount of mental gymnastics.

Either way, I agree completely with the perspective that we need a transitional strategy. I suspect a deprecation would suffice?